index.js

```diff
@@ -57,14 +58,38 @@ Lexer.prototype = {
     if (!value) this.error('ASSERT_FAILED', message);
   },
 
-  assertExpression: function (exp) {
+  assertExpression: function (exp, noThrow) {
     //this verifies that a JavaScript expression is valid
+    var src = 'return (' + exp + ')';
     try {
-      Function('', 'return (' + exp + ')');
+      acorn.parse(src, {
+        ecmaVersion: 6,
+        allowReturnOutsideFunction: true
+      });
     } catch (ex) {
-      var msg = 'Syntax Error: ' + ex.message;
-      this.error('SYNTAX_ERROR', msg);
+      try {
+        acorn.parse(src, {
+          ecmaVersion: 5,
+          allowReturnOutsideFunction: true
+        });
```

Comment:
I don't think we need to add the `ecmaVersion: 5` attempt here.  It's only a very limited set of things that fail to parse in `ecmaVersion: 6` and they're very bad things to write.  Lets force `ecmaVersion: 6` while we're publishing a breaking version anyway.